identity/src/main/java/com/stripe/android/identity/navigation/ConsentFragment.kt

```diff
@@ -50,50 +50,62 @@ internal class ConsentFragment(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View {
-        binding = ConsentFragmentBinding.inflate(inflater, container, false)
-
-        consentViewModel.loadUriIntoImageView(
-            identityViewModel.verificationArgs.brandLogo,
-            binding.merchantLogo
-        )
-
-        return binding.root
+    ) = ComposeView(requireContext()).apply {
+        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+        setContent {
+            val verificationState by identityViewModel.verificationPage.observeAsState(Resource.loading())
+            ConsentScreen(
+                verificationState = verificationState,
+                onMerchantViewCreated = {
+                    consentViewModel.loadUriIntoImageView(
+                        identityViewModel.verificationArgs.brandLogo,
+                        it
+                    )
+                },
+                onSuccess = { verificationPage ->
+                    identityViewModel.updateAnalyticsState { oldState ->
+                        oldState.copy(
+                            requireSelfie = verificationPage.requireSelfie()
+                        )
+                    }
+                },
+                onFallbackUrl = { url ->
+                    Log.e(TAG, "Unsupported client, launching fallback url")
+                    fallbackUrlLauncher.launchFallbackUrl(url)
+                },
+                onError = { throwable ->
+                    Log.e(
+                        TAG,
+                        "Failed to get verificationPage: $throwable"
+                    )
+                    navigateToErrorFragmentWithFailedReason(throwable)
+                },
+                onConsentAgreed = { requireSelfie ->
+                    postVerificationPageDataAndNavigate(
+                        CollectedDataParam(
+                            biometricConsent = true
+                        ),
+                        requireSelfie
+                    )
+                },
+                onConsentDeclined = { requireSelfie ->
+                    postVerificationPageDataAndNavigate(
+                        CollectedDataParam(
+                            biometricConsent = false
+                        ),
+                        requireSelfie
+                    )
```

Comment:
I know these was there before, but a better practice to keep layers separated would be:

1. View (composable) calls `viewModel::onConsentAgreed`
2. ViewModel makes verification page post API call (this already happens that way)
3. ViewModel triggers a state change
  3a. Change affecting the screen, such a permanent error display: something like state.copy(errorMessage: "...")
  3b. One-off events: They should also cause a state change. I've been using a `ViewEffect` [nullable field on the state](https://github.com/stripe/stripe-android/blob/5b628782cfecf14e61007f9c37e9ee30e6c54325/financial-connections/src/main/java/com/stripe/android/financialconnections/FinancialConnectionsSheetState.kt#L18). More about one-off events and anti-patterns [here](https://medium.com/androiddevelopers/viewmodel-one-off-event-antipatterns-16a1da869b95).

Reply:
Thanks for the tips the reason we have the extension `Fragment.postVerificationPageDataAndMaybeSubmit` is that the SDK was set up with a single activity with multiple fragment with NavHostFragment, it needs to have `Fragment.findNavController()` to navigate around fragments. 

This navController will be replaced with the NavHost with composable once all fragments are removed. Will do a refactor with suggested changes when we reach there, added them in [this](https://jira.corp.stripe.com/browse/IDPROD-4581) internal ticket